### PR TITLE
Publish packages automatically after merge

### LINKED ISSUE #378: Publish packages automatically after merge
Currently the publication is manually done by human. We may add a workflow to the CI that does the following things.

- Check for packages that need publishing (by comparing versions of local packages and packages on pub.dev).
- Set up pub.dev credentials (`accessToken` and `refreshToken`).
- Run `pub publish --dry-run` for each package on pull request.
- Run `pub publish` for each package on push.

Basic idea: https://medium.com/evenbit/publishing-dart-packages-with-github-actions-5240068a2f7d

### PR DESCRIPTION
**Pull request by HakkyuKim**
Closes #378.

### For pull requests events:
1. Checks for packages where version field in `pubspec.yaml` changed.
2. Run `pub publish --dry-run` to make CI fail on any errors or warnings. For example, missing a corresponding version entry in `CHANGELOG.md`.

### For push events on master branch:
1. Hold off completing `release` workflow until all other workflows are finished.
2. Run `pub publish --dry-run` again just in case.
3. Publish the changed packages identified in the pushed PR.

## Prerequisite
The content of the `credential.json`, which is created when running `dart pub login`, must be added to [Github repository secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) with name "PUB_CREDENTIALS". 

## Limitation
New packages must be published manually because pub.dev current doesn't support publishing a new package directly to a verified publisher, See https://dart.dev/tools/pub/publishing#publishing.

## TODO
- [x] Create release workflow similar to https://github.com/flutter/plugins/blob/main/.github/workflows/release.yml.
- [x] Fix step "Wait on all tests", it's not waiting for other workflows to be finished. -> The behavior was expected in a pull request event, I've confirmed it works on master branch. See https://github.com/lewagon/wait-on-check-action/issues/25 for details.
- [x] Test: pull_request event:
    - [x] Changes without upgrading version. -> Expected: CI success.
    - [x] Upgrading version without CHANGELOG entry. -> Expected: CI failure.
    - [x] Positive case.
- [ ] Test: push event on master branch:
    - [ ] Check whether `release` is the last workflow to be finished.
    - [ ] Check whether the upgraded package is released.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (swift-kim) [inline .github/workflows/release.yml:14]: The flutter-tizen tool is unused. Just using `subosito/flutter-action@v1` will do.

Comment (HakkyuKim) [reply .github/workflows/release.yml]: Done.

**Comment by HakkyuKim** [comment]
We can put off setting PUB_CREDENTIALS to Github secrets and first see if the workflow runs correctly on the master branch, if it does, we can rerun the job after adding PUB_CREDENTIALS. I see `google_maps_flutter_tizen` isn't published for `0.1.2`, shall I try with this?

**Comment by HakkyuKim** [comment]
The flutter team [allowed neutral conclusion due to Cirrus CI](https://github.com/flutter/plugins/pull/4321), it shouldn't be a problem on our side.

File: .github/workflows/release.yml
Comment (swift-kim) [inline .github/workflows/release.yml:45]: You can safely remove this dry run.

File: .github/workflows/release.yml
Comment (swift-kim) [inline .github/workflows/release.yml:48]: What's different from the above `--base-sha=$(git rev-parse HEAD~)`?

File: .github/workflows/release.yml
Comment (swift-kim) [inline .github/workflows/release.yml:1]: Please capitalize and add a trailing newline for consistency.

```suggestion
name: Release

```

File: tools/lib/src/main.dart
Comment (swift-kim) [inline tools/lib/src/main.dart:14]:
```suggestion
import 'publish_plugin_command.dart';
```

**Review by swift-kim — COMMENTED** [review]
Otherwise overall looks good. Thank you!

**Comment by swift-kim** [comment]
@HakkyuKim You can test the push event with this PR: https://github.com/flutter-tizen/plugins/pull/392

File: .github/workflows/release.yml
Comment (HakkyuKim) [reply .github/workflows/release.yml:1]: Done!

File: .github/workflows/release.yml
Comment (HakkyuKim) [reply .github/workflows/release.yml:45]: Done

File: .github/workflows/release.yml
Comment (HakkyuKim) [reply .github/workflows/release.yml:48]: Oh, they aren't different in terms of program execution. I'll use `HEAD~`, thanks!

File: tools/lib/src/main.dart
Comment (HakkyuKim) [reply tools/lib/src/main.dart:14]: Done

**Comment by HakkyuKim** [comment]
@swift-kim 
Okay, I'll merge #392 after this PR is merged.

File: .github/workflows/release.yml
Comment (HakkyuKim) [reply .github/workflows/release.yml]: #392 and #387 won't get a chance to --dry-run without rebasing this PR but I've checked both PR locally and they passed the --dry-run command.

**Review by swift-kim — APPROVED** [review]

**Comment by swift-kim** [comment]
https://github.com/flutter-tizen/plugins/runs/6586016243

The publication was successful after updating PUB_CREDENTIALS.

**Pull request merged**
